### PR TITLE
vendor: Bump kubernetes dependencies to 1.20.2

### PR DESCRIPTION
**Pull request by adamwg**
As part of this change, we have to update our main to use k8s.io/cloud-provider/app, since the old version of it has now been removed from the Kubernetes tree.

### CODE REVIEW

File: cloud-controller-manager/cmd/digitalocean-cloud-controller-manager/main.go [hunk 1]
```diff
@@ -23,14 +23,21 @@ import (
...
 
 	"github.com/digitalocean/digitalocean-cloud-controller-manager/cloud-controller-manager/do"
 	"github.com/spf13/pflag"
 )
 
 func main() {
-	command := app.NewCloudControllerManagerCommand()
+	opts, err := options.NewCloudControllerManagerOptions()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to construct options: %v\n", err)
```
Comment (timoreimann) [inline cloud-controller-manager/cmd/digitalocean-cloud-controller-manager/main.go:37]: Is it correct that we emit an error message but do not terminate the program?

**Review by timoreimann — COMMENTED** [review]
One quick question, and maybe the ask to append the change log briefly. Otherwise all good. 👍 

Glad to see we can finally stop vendoring k/k. 🎉

**Review by timoreimann — APPROVED** [review]
(LGTM'ing to save one round trip.)

File: cloud-controller-manager/cmd/digitalocean-cloud-controller-manager/main.go (hunk 1 above)
Comment (adamwg) [reply cloud-controller-manager/cmd/digitalocean-cloud-controller-manager/main.go:37]: Oops, good catch. Fixed.

**Review by timoreimann — APPROVED** [review]

**Comment by adamwg** [comment]
This doesn't quite work properly due to https://github.com/kubernetes/cloud-provider/issues/44. Let's revisit this once the upstream situation is clearer.

**Pull request closed without merging**